drivers/gpio/gpio_mcux.c

```diff
@@ -275,182 +275,97 @@ static const struct gpio_driver_api gpio_mcux_driver_api = {
 	.disable_callback = gpio_mcux_disable_callback,
 };
 
-#ifdef CONFIG_GPIO_MCUX_PORTA
-static int gpio_mcux_porta_init(struct device *dev);
-
-static const struct gpio_mcux_config gpio_mcux_porta_config = {
-	.common = {
-		.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_NGPIOS(DT_INST_0_NXP_KINETIS_GPIO_NGPIOS),
-	},
-	.gpio_base = (GPIO_Type *) DT_ALIAS_GPIO_A_BASE_ADDRESS,
-	.port_base = PORTA,
-#ifdef DT_ALIAS_GPIO_A_IRQ_0
-	.flags = GPIO_INT_ENABLE,
+#define GPIO_MCUX_IRQ_INIT(n)							\
+	do {									\
+		IRQ_CONNECT(DT_INST_## n ##_NXP_KINETIS_GPIO_IRQ_0,		\
+			    DT_INST_## n ##_NXP_KINETIS_GPIO_IRQ_0_PRIORITY,	\
+			    gpio_mcux_port_isr, DEVICE_GET(gpio_mcux_port##n), 0);\
+										\
+		irq_enable(DT_INST_## n ##_NXP_KINETIS_GPIO_IRQ_0);		\
+	} while (0)
```

Comment:
why is this inside a do-while loop?

Reply:
checkpatch considers it an error if you dont.